xmtp_mls/src/client.rs

```diff
@@ -370,13 +356,48 @@ where
         addresses: &[String],
     ) -> Result<Vec<Option<String>>, ClientError> {
         let sanitized_addresses = sanitize_evm_addresses(addresses)?;
-        let mut results = self
-            .api_client
-            .get_inbox_ids(sanitized_addresses.clone())
-            .await?;
-        let inbox_ids: Vec<Option<String>> = sanitized_addresses
+        let conn = self.store().conn()?;
+
+        let local_results: Vec<WalletEntry> = conn.fetch_wallets_list_with_key(&sanitized_addresses)?;
+
+        let mut results: HashMap<String, String> = local_results
             .into_iter()
-            .map(|address| results.remove(&address))
+            .map(|entry| (entry.wallet_address.clone(), entry.inbox_id.clone()))
+            .collect();
+
+        let missing_addresses: Vec<String> = sanitized_addresses
+            .iter()
+            .filter(|address| !results.contains_key(*address))
+            .cloned()
+            .collect();
+
+        if missing_addresses.is_empty() {
+            let inbox_ids: Vec<Option<String>> = sanitized_addresses
+                .iter()
+                .map(|address| results.get(address).cloned())
+                .collect();
+            return Ok(inbox_ids);
+        }
+
+        let web_results = self.api_client.get_inbox_ids(missing_addresses.clone()).await?;
+
+        for (address, inbox_id) in &web_results {
+            results.insert(address.clone(), inbox_id.clone()).unwrap_or_default();
+        }
+
+        for (address, inbox_id) in &web_results {
```

Comment:
We can combine these, and remove the borrow in the `web_results` we're iterating on.